pkg/state/infra.go

```diff
@@ -86,16 +88,17 @@ type infraClient struct {
 	resyncInterval time.Duration
 
 	// Guards all fields in the internal state of the client
-	lock *sync.Mutex
+	lock           *sync.Mutex
 }
 
-func NewInfra(routerFactory routerStateFunc, brokerFactory brokerStateFunc, clock Clock) *infraClient {
+func NewInfra(i *v1.MessagingInfrastructure, routerFactory routerStateFunc, brokerFactory brokerStateFunc, clock Clock) *infraClient {
 	// TODO: Make constants and expand range
 	portmap := make(map[int]*string, 0)
 	for i := 40000; i < 40100; i++ {
 		portmap[i] = nil
 	}
 	client := &infraClient{
+		infrastructure:      i,
```

Comment:
I see that only the name and namespace of the infrastructure is actually used. I wonder if it would be better to store just that, because this is not synced, so the rest of the infra could get stale if the user changes the spec for instance. Just to future proofing myself from looking at the infrastructure and thinking i can pull some data from the spec.

Reply:
Thanks, I hadn't considered a staleness problem.  Fixed.